godot-core/src/registry/callbacks.rs

```diff
@@ -25,6 +25,14 @@ use crate::private::{handle_panic, PanicPayload};
 use crate::registry::plugin::ErasedDynGd;
 use crate::storage::{as_storage, InstanceStorage, Storage, StorageRefCounted};
 
+#[cfg(since_api = "4.4")]
+unsafe fn notify_postinit(base_ptr: sys::GDExtensionObjectPtr) {
+    // Should notify it with a weak pointer, during `NOTIFICATION_POSTINITIALIZE`, ref-counted object is not yet fully-initialized.
+    let mut obj = Gd::<Object>::from_obj_sys_weak(base_ptr);
+    obj.notify(crate::classes::notify::ObjectNotification::POSTINITIALIZE);
+    std::mem::forget(obj);
+}
```

Comment:
See https://github.com/godot-rust/gdext/pull/1211#discussion_r2284367189

> Additionally, is `mem::forget()` truly correct here? `Gd`/`RawGd` has other data members than just the object pointer... most of them are currently `Copy` but we can't rely on that, and they might leak 🤔
> 
> Maybe there should be a dedicated method `RawGd::drop_weak(self)` which doesn't invoke dec-ref...

Reply:
For now, I'm also fine with `Gd::drop_weak(self)` which internally calls `mem::forget()`. At least that gives us a place to make this explicit, which could be changed in the future if we have fields that need dealloc.

Could you add `Gd::drop_weak()` (crate-private) and also substitute the following occurrence with it?

https://github.com/godot-rust/gdext/blob/8526478531fae43c745e4e76a2efd121b7044c00/godot-core/src/obj/gd.rs#L729-L730